### ISSUE TITLE
More reliable way to sync actual and visible values

### ISSUE THREAD

**Issue by chemerisuk** [chemerisuk/better-dateinput-polyfill#43]
It looks like triggering `change` event sometimes is not obvious. Need to investigate if it's possible to use mutation events/`propertychange` to achieve the same goal.

Related issues:
https://github.com/chemerisuk/better-dateinput-polyfill/issues/34
https://github.com/chemerisuk/better-dateinput-polyfill/issues/14

**Comment by Orangetronic** [comment]
Related: if the polyfill instantiates after a value has been set on a date field, it clears the value. So an example use-case might be an edit form, where a form loads with a particular set of values pre-set. 

Right now the best way I've found to solve this is to set a timeout after page load, re-set values, and fire change events, which is all pretty clumsy. 

Won't be able to get to it this weekend, but I'd be up for having a go at adding a 'takes existing field value if correctly formatted' feature at some point.

**Comment by vertti** [comment]
Hoping to get this polyfill to work with React. But react doesn't see the values input with this poly. This seems related.

**Comment by tomascharad** [comment]
Hi guys, it would be very nice if you could make your poly compatible with react.
Many thanks!
Let me know if I can help.

**Comment by chemerisuk** [comment]
Okay, I've implemented PoC for a new approach to sync visible and serializable values https://github.com/chemerisuk/better-dateinput-polyfill/commit/0a9867ecd9e3fc1485a63326ebef6ff36a632524. This one uses overriding of the `value` property for the input and does not rely on any events. The logic is much cleaner now. Checked in Safari/Chrome/Firefox - works smoothly, should work in IE9+. I'll probably remove IE8 support in the next major version.

Any help with cross browser (and cross framework) testing is appreciated.
